modules/hub/src/api/PaymentsApiService.ts

```diff
@@ -147,12 +147,15 @@ export class PaymentsApiServiceHandler {
         },
       )
       return res.sendStatus(400)
-    }
+    } 
 
     const result = await this.paymentsService.doRedeem(user, secret)
     if (result.error != false) {
       LOG.warn(result.msg)
-      return res.send(400).json(result.msg)
+      // @ts-ignore
+      // TODO: wtf? it works, but doesnt compile
+      // are the express types out of date somehow?
+      return res.send(500, result.msg)
```

Comment:
We should throw 400 or something different than 500 here, since it's not an uncaught exception.

Reply:
https://expressjs.com/en/4x/api.html#res.status

See above for documented way to do what you're doing.

Reply:
We should add this everywhere lol.